src/DayPicker.tsx

```diff
@@ -256,6 +263,118 @@ export function DayPicker(props: DayPickerProps) {
+      months[0].date,
+      previousMonths.current[0].date
+    );
+    const slideLeft = dateLib.isAfter(
+      months[0].date,
+      previousMonths.current[0].date
+    );
+    previousMonths.current = months;
+    if (rootRef.current) {
+      const previousMonthElement =
+        rootElementSnapshot.current?.querySelector(`[data-month]`);
+      rootElementSnapshot.current = rootRef.current.cloneNode(
+        true
+      ) as HTMLElement;
+      if (isSameMonth) {
+        return;
+      }
+
+      const currentMonthElement = rootRef.current.querySelector(`[data-month]`);
+      if (
+        currentMonthElement &&
+        currentMonthElement instanceof HTMLElement &&
+        previousMonthElement &&
+        previousMonthElement instanceof HTMLElement
+      ) {
+        // animate new displayed month
+        currentMonthElement.style.overflow = "hidden";
+        const monthCaptionElement =
+          rootRef.current.querySelector(`[data-month-caption]`);
+        if (monthCaptionElement && monthCaptionElement instanceof HTMLElement) {
+          monthCaptionElement.style.animation = "fadeIn 0.3s forwards";
+        }
+
+        const weeksElement = rootRef.current.querySelector(`[data-weeks]`);
+        if (weeksElement && weeksElement instanceof HTMLElement) {
+          weeksElement.style.animation = slideLeft
+            ? "slideInRight 0.3s forwards"
+            : "slideInLeft 0.3s forwards";
+        }
+        // animate new displayed month end
+        const cleanUp = () => {
+          if (
+            monthCaptionElement &&
+            monthCaptionElement instanceof HTMLElement
+          ) {
+            monthCaptionElement.style.animation = "";
+          }
+          if (weeksElement && weeksElement instanceof HTMLElement) {
+            weeksElement.style.animation = "";
+          }
+          currentMonthElement.style.overflow = "";
+          if (currentMonthElement.contains(previousMonthElement)) {
+            currentMonthElement.removeChild(previousMonthElement);
+          }
+        };
+
+        // animate old displayed month
+        previousMonthElement.style.pointerEvents = "none";
+        previousMonthElement.style.position = "absolute";
+        previousMonthElement.style.overflow = "hidden";
+        previousMonthElement.setAttribute("aria-hidden", "true");
+
+        const previousWeekdaysElement =
+          previousMonthElement.querySelector(`[data-weekdays]`);
+        if (
+          previousWeekdaysElement &&
+          previousWeekdaysElement instanceof HTMLElement
+        ) {
+          previousWeekdaysElement.style.opacity = "0";
+        }
+
+        const previousWeeksElement =
+          previousMonthElement.querySelector(`[data-weeks]`);
+        if (
+          previousWeeksElement &&
+          previousWeeksElement instanceof HTMLElement
+        ) {
+          previousWeeksElement.style.animation = slideLeft
+            ? "slideOutLeft 0.3s forwards"
```

Comment:
In a future version, the transition could be customizable via props. 

It would be useful to try a up/down animation (as opposed to left-right) to see how different transitions would actually work, or which challenges could they pose to the developer.

Reply:
Indeed good point,
I'll work on an API that allows it already and will test it by trying to write a custom components example that has the animation up and down, writing such example is a good validation of the API allowing customization.